plugins/inventory/azure_rm.py

```diff
@@ -354,8 +367,10 @@ def _get_hosts(self):
                 continue
             self.inventory.add_host(inventory_hostname)
             # FUTURE: configurable default IP list? can already do this via hostvar_expressions
-            self.inventory.set_variable(inventory_hostname, "ansible_host",
-                                        next(chain(h.hostvars['public_ipv4_address'], h.hostvars['private_ipv4_addresses']), None))
+            if 'public_ipv4_address' in h.hostvars and \
```

Comment:
i think it should be 'or' because if there is no public ip, we still want to use the private one as default for ansible_host
(otherwise ansible_host is not set)

Reply:
correction : the if should be removed to always set variable ansible_host
but hostvars public_ipv4_address & private_ipv4_addresses should exists on the Arc object